zipline/data/hdf5_daily_bars.py

```diff
@@ -466,41 +466,63 @@ def load_raw_arrays(self,
             (minutes in range, sids) with a dtype of float64, containing the
             values for the respective field over start and end dt range.
         """
-        self._validate_assets(assets)
         self._validate_timestamp(start_date)
         self._validate_timestamp(end_date)
 
         start = start_date.asm8
         end = end_date.asm8
 
-        sid_selector = self.sids.searchsorted(assets)
+        sid_selector, out_buf_indexer = self._make_sid_indexers(assets)
+
         date_slice = self._compute_date_range_slice(start, end)
 
-        nrows = date_slice.stop - date_slice.start
-        ncols = len(self.sids)
+        n_dates = date_slice.stop - date_slice.start
+        n_query_sids = len(assets)
+        n_valid_sids = len(self.sids)
 
-        buf = np.zeros((ncols, nrows), dtype=np.uint32)
+        read_buf = np.zeros((n_valid_sids, n_dates), dtype=np.uint32)
+        out_buf = np.zeros((n_query_sids, n_dates), dtype=np.uint32)
 
         out = []
         for column in columns:
-            # Zero the buffer to prepare it to receive new data.
-            buf.fill(0)
+            # Zero the buffers to prepare to receive new data.
+            read_buf.fill(0)
+            out_buf.fill(0)
 
             dataset = self._country_group[DATA][column]
 
             dataset.read_direct(
-                buf,
+                read_buf,
                 np.s_[:, date_slice.start:date_slice.stop],
```

Comment:
Not changed here, but reading this again I think this might just be `np.s_[:, date_slice]`?